### PR TITLE
Configurable default HealthCheck timeout

### PR DESCRIPTION
**Pull request by plamen-bardarov**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
Make the declarative Liveness Healthcheck default timeout configurable.

Backward Compatibility
---------------
Breaking Change? **No**

### CODE REVIEW

File: depot/transformer/transformer_test.go [hunk 1]
```diff
@@ -1400,12 +1402,42 @@ var _ = Describe("Transformer", func() {
...
+						Context("and the default declarative healthcheck timeout is not set in the spec", func() {
+							BeforeEach(func() {
+								var emptyJsonTime durationjson.Duration
+								declarativeHealthCheckTimeout := time.Duration(emptyJsonTime)
+
+								// This option will override the previously configured default timeout
+								options = append(options, transformer.WithDeclarativeHealthchecks(declarativeHealthCheckTimeout))
+							})
+
+							It("uses the hardcoded 1s default for the timeout", func() {
```
Comment (ameowlia) [inline depot/transformer/transformer_test.go:1421]:
```suggestion
							It("uses the 1s default for the timeout", func() {
```

File: depot/transformer/transformer.go [hunk 2]
```diff
@@ -820,9 +821,14 @@ func (t *transformer) transformReadinessCheckDefinition(
 }
 
 func (t *transformer) applyCheckDefaults(timeout int, interval time.Duration, path string) (int, time.Duration, string) {
-	if timeout == 0 {
+	if timeout <= 0 {
+		timeout = int(t.declarativeHealthCheckDefaultTimeout / time.Millisecond)
+	}
+	if timeout <= 0 {
```
Comment (ameowlia) [inline depot/transformer/transformer.go:827]: Can you add a test for this case where the declarativeHealthCheckDefaultTimeout is set to a negative number?

File: depot/transformer/transformer.go [hunk 3]
```diff
@@ -820,9 +821,14 @@ func (t *transformer) transformReadinessCheckDefinition(
 }
 
 func (t *transformer) applyCheckDefaults(timeout int, interval time.Duration, path string) (int, time.Duration, string) {
-	if timeout == 0 {
```
Comment (ameowlia) [inline depot/transformer/transformer.go:823]: Can you move this logic about defaults to the initializer package where the config is read in? That way the t.declarativeHealthCheckDefaultTimeout property is always correct.

**Review by ameowlia — CHANGES_REQUESTED** [review]

File: depot/transformer/transformer.go (hunk 2 above)
Comment (ameowlia) [reply depot/transformer/transformer.go:827]: Also maybe we should log in this case? Something like: you gave us an invalid value for X we are setting it to Y instead.

File: depot/transformer/transformer.go (hunk 3 above)
Comment (plamen-bardarov) [reply depot/transformer/transformer.go:823]: Moved it to the configuration option, which is still called only during initialization.

File: depot/transformer/transformer.go (hunk 2 above)
Comment (plamen-bardarov) [reply depot/transformer/transformer.go]: I think its better if we only accept valid values for this field instead of relying on this default if the job is misconfigured so I added it to the validation method. However since we receive an already validated object from the rep, testing this is not possible in the executor 😕. I left the <=0 check just in case something changes.
